### DISCUSSION TITLE
Selecting  A Subset of Fields in an embedded Array

### DISCUSSION THREAD

**Discussion by MDrooker** [vazco/uniforms#1151] [Q&A] [answered]
I have a model that looks like
```
{
    "creatoruuid": "johndoe",
    "shareLinks": [
        {
            "version": 1,
            "name": "default",
            "effectiveStartDate": "2022-07-24T13:41:00.000Z",
            "effectiveEndDate": "2022-12-31T13:41:00.000Z",
            "metadata": [],
            "reviewData": [],
            "isActive": true,
            "isDeleted": false,
            "isLocked": false,
        }
    ]
}
```

I only want to create a form that allows a subset of fields in the shareLinks array.  
So for example I just want to allow the "version" and "name" fields to be presented to the user.

I tried 
```
 <AutoFields fields={['shareLinks.version', 'shareLinks.name']} />
```
But got the error-
```
Invariant Violation: Field not found in schema: "shareLinks.version"
```
**Question-
How do I index into an array and just subset of fields**

**Comment by radekmie** [comment, marked as answer]
Hi @MDrooker. Unfortunately, it's not possible right now. You can use `$` as "any index", but it won't let you group the fields whatsoever. The best I can think of right now, would be to do something like this:

```tsx
<AutoForm {...}>
  <AutoFields fields={['a', 'b']} />
  <ListField name="c">
    <ListItemField name="$">
      <AutoFields fields={['x', 'y']} />
    </ListItemField>
  </ListField>
  <AutoFields fields={['d', 'e']} />
</AutoForm>
```

_(That's just a sketch; the exact structure depends on your schema.)_

---

Another option would be to create a subschema (a schema with a subset of fields), only for this form. It may be easier, depending on your case.
